tag/validator.go

```diff
@@ -0,0 +1,103 @@
+package tag
+
+import (
+	"fmt"
+
+	ut "github.com/go-playground/universal-translator"
+	v "github.com/go-playground/validator/v10"
+	"github.com/odpf/columbus/tag/validator"
+)
+
+// newValidator initializes validator for tag
+func newValidator() validator.Validator {
+	v, err := validator.NewBuilder().
+		WithTranslations([]validator.Translation{
+			{
+				Tag:      "required",
+				Message:  "cannot be empty",
+				Override: true,
+			},
+			{
+				Tag:      "min",
+				Message:  "must be at least {0}",
+				Override: true,
+				TranslationFunc: func(t ut.Translator, fe v.FieldError) string {
+					output, _ := t.T(fe.Tag(), fe.Param())
+					return output
+				},
+			},
+		}).
+		Build()
+	if err != nil {
+		panic(err)
+	}
+
+	return v
+}
+
+// newTemplateValidator initializes validator for tag template
+func newTemplateValidator() validator.Validator {
+	v, err := validator.NewBuilder().
+		WithStructValidations([]validator.StructValidation{
+			{
+				Type: Template{},
+				Func: func(sl v.StructLevel) {
+					template := sl.Current().Interface().(Template)
+					for i, field := range template.Fields {
```

Comment:
This could lead to nil pointer panic if the interface above failed to be casted. Better to check if casting is ok or not.

Reply:
yeah same, didn't want to change Dexter logic before, but will try to see if we can do that.

Reply:
updated